storage/Utils/CommentedConfigFile.h

```diff
@@ -294,7 +297,7 @@ class CommentedConfigFile: private boost::noncopyable
     /**
      * Return entry no. 'index' or 0 if 'index' is out of range.
      **/
-    Entry * get_entry( int index );
+    Entry * get_entry( int index ) const;
```

Comment:
That's a philosophical question... this 'const' promises it won't modify the CommentedConfigFile object. That's not quite the same as promising not to modify one of the objects it keeps pointers to. That's how large libs like Qt do it all the time; getting a pointer to some items is a const operation, even though the item itself may be modified by a non-const pointer.

If you see it that strictly, there are almost no real const methods left because this non-constness here will spread out like a disease.